### PR TITLE
NeXus type hardening

### PR DESCRIPTION
**Pull request by graeme-winter**
- If scalar, don't fail
- Cast to numpy str for comparison

### CODE REVIEW

**Comment by graeme-winter** [comment]
Reviewing in light of test failures now

File: format/nexus.py [hunk 1]
```diff
@@ -287,10 +287,13 @@ def visit(self, nx_file, depends_on):
...
-                    is_scan_axis = True
-                else:
+                v = item[...]
+                try:
+                    if min(v) < max(v):
+                        is_scan_axis = True
+                    else:
+                        is_scan_axis = False
+                except TypeError:
                     is_scan_axis = False
```
Comment (benjaminhwilliams) [inline format/nexus.py:297]: I suggest this, which is safe if your dataset is scalar or iterable:
```suggestion
                v = item[...]
                if v.min() < v.max():
                    is_scan_axis = True
                else:
                    is_scan_axis = False
```

File: format/nexus.py [hunk 2]
```diff
@@ -1139,8 +1142,11 @@ def find_goniometer_rotation(obj):
         if o.attrs["transformation_type"] == numpy.string_("rotation"):
             # if this is changing, assume is scan axis
             v = o[()]
```
Comment (benjaminhwilliams) [inline format/nexus.py:1144]:
```suggestion
            v = o[...]
```

File: format/nexus.py [hunk 3]
```diff
@@ -1139,8 +1142,11 @@ def find_goniometer_rotation(obj):
...
-            if min(v) < max(v):
-                return o
+            try:
+                if min(v) < max(v):
+                    return o
+            except TypeError:
+                continue
```
Comment (benjaminhwilliams) [inline format/nexus.py:1149]:
```suggestion
            if v.min() < v.max():
                return o
```

**Review by benjaminhwilliams — CHANGES_REQUESTED** [review]
If the values are always guaranteed to be numbers, I think my suggestion is safe.

**Comment by ndevenish** [comment]
Updated to resolve conflicts. Since #263 was merged into master, "Cast to numpy str for comparison" is no longer relevant.

**Pull request merged**